presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java

```diff
@@ -401,6 +436,19 @@ public OperatorStats add(Iterable<OperatorStats> operators)
             memoryReservation += operator.getUserMemoryReservation().toBytes();
             revocableMemoryReservation += operator.getRevocableMemoryReservation().toBytes();
             systemMemoryReservation += operator.getSystemMemoryReservation().toBytes();
+
+            if (operator.getPeakUserMemoryReservation().toBytes() > peakUserMemory) {
+                peakUserMemory = operator.getPeakUserMemoryReservation().toBytes();
```

Comment:
why not use `peakUserMemory = max(peakUserMemory, operator.getPeakUserMemoryReservation().toBytes())`?